fastjet/internal/delaunay/delaunay.go

```diff
@@ -87,7 +87,18 @@ func (d *Delaunay) Insert(p *Point) (updatedNearestNeighbor []*Point) {
 // whose nearest neighbor changed due to the removal. The slice may contain
 // duplicates.
 func (d *Delaunay) Remove(p *Point) (updatedNearestNeighbor []*Point) {
-	panic(fmt.Errorf("delaunay: Remove not implemented"))
+	if len(p.adjacentTriangles) < 3 {
+		panic(fmt.Errorf("delaunay: can't remove point %v, not enough adjacent triangles", p))
```

Comment:
Yes, for the hierarchical method the only points on the outside are the root points, which cannot be removed.